src/schema/submissions.ts

```diff
@@ -0,0 +1,176 @@
+  }
+`;
+
+const submissionLimit = parseInt(
+  process.env.SCOUT_SUBMISSION_LIMIT || DEFAULT_SUBMISSION_LIMIT,
+);
+
+export const submissionAccessThreshold = parseInt(
+  process.env.SCOUT_SUBMISSION_ACCESS_THRESHOLD ||
+    DEFAULT_SUBMISSION_ACCESS_THRESHOLD,
+);
+
+const hasSubmissionAccess = (user: User) =>
+  user.reputation >= submissionAccessThreshold;
+
+const getSubmissionsToday = (con: Connection, userTimezone?: string) => {
+  const timezone = userTimezone ?? 'utc';
+  const nowTimezone = `timezone('${timezone}', now())`;
+  const atTimezone = `at time zone '${timezone}'`;
+
+  return con.getRepository(Submission).find({
+    where: `date_trunc('day', "createdAt")::timestamptz ${atTimezone} = date_trunc('day', ${nowTimezone})::timestamptz`,
+  });
+};
+
+export const resolvers: IResolvers<unknown, Context> = traceResolvers({
+  Query: {
+    submissionAvailability: async (
+      _,
+      __,
+      ctx,
+    ): Promise<GQLSubmissionAvailability> => {
+      const user = await ctx.getRepository(User).findOne({ id: ctx.userId });
+      if (!user) {
+        return {
+          limit: submissionLimit,
+          hasAccess: false,
+          todaySubmissionsCount: 0,
+        };
+      }
+
+      const submissionsToday = await getSubmissionsToday(
+        ctx.con,
+        user.timezone,
+      );
+
+      return {
+        limit: submissionLimit,
+        todaySubmissionsCount: submissionsToday.length,
+        hasAccess: hasSubmissionAccess(user),
+      };
+    },
+  },
+  Mutation: {
+    submitArticle: async (
+      _,
+      { url }: GQLArticleSubmission,
+      ctx,
+      info,
+    ): Promise<GQLSubmitArticleResponse> => {
+      const user = await ctx.getRepository(User).findOne({ id: ctx.userId });
+
+      if (!user || !hasSubmissionAccess(user)) {
+        return { result: 'reject', reason: 'Access denied' };
+      }
+
+      const submissionRepo = ctx.con.getRepository(Submission);
+      const submissionsToday = await getSubmissionsToday(
+        ctx.con,
+        user.timezone,
+      );
+
+      if (submissionsToday.length >= submissionLimit) {
+        return { result: 'reject', reason: 'Submission limit reached' };
+      }
+
+      if (!isValidHttpUrl(url)) {
+        return { result: 'reject', reason: 'invalid URL' };
+      }
+
+      const existingPost = await getPostByUrl(url, ctx, info);
+      if (existingPost) {
+        if (existingPost.deleted) {
+          return { result: 'reject', reason: 'post is deleted' };
+        }
+        return { result: 'exists', post: existingPost };
+      }
+
+      const existingSubmission = await submissionRepo.findOne({ url });
+
+      if (existingSubmission) {
+        return {
+          result: 'reject',
+          reason: `Article has been submitted already! Current status: ${existingSubmission.status}`,
+        };
+      }
+
+      const submission = await submissionRepo.save(
+        submissionRepo.create({ url, userId: ctx.userId }),
+      );
```

Comment:
I think if `url` is marked as unique you can merge it down to a single save statement and if there will be a unique violation an error will be thrown and you can check the error number to confirm it's a unique violation